### PR TITLE
Update `remark`, other dependencies

### PR DESCRIPTION
**Pull request by wooorm**
I thought I’d update remark because it includes the new `meta` prop, which simplifies the code here: https://github.com/mdx-js/mdx/blob/6c446d69beaf04eabe708b90a6ac9634fd1c02af/packages/mdx/mdx-ast-to-mdx-hast.js

I also updated the other dependencies/dev-dependencies.

### CODE REVIEW

File: packages/mdx/mdx-ast-to-mdx-hast.js [hunk 1]
```diff
@@ -23,18 +23,18 @@ function mdxAstToMdxHast() {
       code(h, node) {
         const langRegex = /^[^ \t]+(?=[ \t]|$)/
```
Comment (johno) [inline packages/mdx/mdx-ast-to-mdx-hast.js]: Think we can ✂️ this as well now?

File: package.json [hunk 2]
```diff
@@ -30,7 +30,7 @@
   },
   "dependencies": {
     "@rebass/mdx": "^1.0.0-1",
-    "styled-components": "^3.3.3"
+    "styled-components": "^4.0.0"
```
Comment (johno) [inline package.json]: I'll need to make an upstream change to `x0` before bumping `styled-components` version so we might want to skip this update as part of this PR. It's on my todo list today to get this fixed.

Comment (wooorm) [reply package.json]: OK, perfect, ping back here when you’re done!

Comment (wooorm) [reply package.json]: Or we can delay, whatever works?

Comment (johno) [reply package.json]: If you want to keep this pinned at 3 you can merge this and I can PR in this update separately (I don't want to be a blocker, currently on the road with spotty internet access).

Comment (wooorm) [reply package.json]: Downgraded! I believe this is ready now, and could be batched up with the docs changes as a new release!

**Pull request merged**